src/main/java/com/coveros/selenified/element/check/Contains.java

```diff
@@ -0,0 +1,281 @@
+     * If the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedClass - the expected class value
+     */
+    void clazz(String expectedClass);
+
+    default String checkClazz(String expectedClass, double waitFor, double timeTook) {
+        // record the action
+        getOutputFile().recordAction(getElement().prettyOutput() + " containing class <b>" + expectedClass + "</b>", waitFor);
+        // get the value
+        String actualClass = getElement().get().attribute(CLASS);
+        // record the result
+        if (actualClass == null || !actualClass.contains(expectedClass)) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + CLASSVALUE + actualClass + "</b>", timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + CLASSVALUE + actualClass + "</b>, which contains <b>" +
+                    expectedClass + "</b>", timeTook, Success.PASS);
+        }
+        return actualClass;
+    }
+
+    /**
+     * Verifies that the element contains the provided expected attribute. If
+     * the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedAttribute - the attribute to check for
+     */
+    void attribute(String expectedAttribute);
+
+    default String[] checkAttribute(String expectedAttribute, double waitFor, double timeTook) {
+        // record the action and get the attributes
+        String[] allAttributes = getAttributes(expectedAttribute, "with", waitFor);
+        // record the result
+        if (allAttributes == null || !Arrays.asList(allAttributes).contains(expectedAttribute)) {
+            getOutputFile().recordActual(
+                    getElement().prettyOutputStart() + " does not contain the attribute of <b>" + expectedAttribute + "</b>" +
+                            ONLYVALUE + Arrays.toString(allAttributes) + "</b>", timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + " contains the attribute of <b>" + expectedAttribute + "</b>",
+                    timeTook, Success.PASS);
+        }
+        return allAttributes;
+    }
+
+    /**
+     * Verifies that the element's text contains the provided expected text. If
+     * the element isn't present, this will constitute a failure, same as a
+     * mismatch. This information will be logged and recorded, with a screenshot
+     * for traceability and added debugging support.
+     *
+     * @param expectedText the expected value of the element
+     */
+    void text(String expectedText);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Rename method "text" to prevent any misunderstanding/clash with field "TEXT" defined in interface "com.coveros.selenified.element.check.Check". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1845)